src/drivers/wayland/lv_wayland.c

```diff
@@ -117,11 +116,12 @@ struct input {
         lv_indev_state_t state;
     } keyboard;
 
-    struct {
-        uint32_t x;
-        uint32_t y;
-        lv_indev_state_t state;
-    } touch;
+#if LV_USE_GESTURE_RECOGNITION
+    lv_indev_touch_data_t touches[10];
+    uint8_t touch_event_cnt;
+    uint8_t primary_id;
+    lv_indev_gesture_recognizer_t recognizer;
+#endif
```

Comment:
Does Wayland normal touch still work with `LV_USE_GESTURE_RECOGNITION` disabled? Or does Wayland depend on it being enabled? If it's required, then all the guards could go away and a single `#if !LV_USE_GESTURE_RECOGNITION ... #error ... #endif` could go at the top.